src/main/java/org/opensearch/knn/jni/JNIService.java

```diff
@@ -13,16 +13,99 @@
 
 import org.apache.commons.lang.ArrayUtils;
 import org.opensearch.common.Nullable;
+import org.opensearch.knn.common.KNNConstants;
 import org.opensearch.knn.index.IndexUtil;
 import org.opensearch.knn.index.query.KNNQueryResult;
 import org.opensearch.knn.index.util.KNNEngine;
 
+import java.util.Locale;
 import java.util.Map;
 
 /**
  * Service to distribute requests to the proper engine jni service
  */
 public class JNIService {
+    /**
+     * Initialize an index for the native library. Takes in numDocs to
+     * allocate the correct amount of memory.
+     *
+     * @param numDocs number of documents to be added
+     * @param dim dimension of the vector to be indexed
+     * @param parameters parameters to build index
+     * @param knnEngine knn engine
+     * @return address of the index in memory
+     */
+    public static long initIndexFromScratch(long numDocs, int dim, Map<String, Object> parameters, KNNEngine knnEngine) {
+        if (KNNEngine.FAISS == knnEngine) {
```

Comment:
Jni Service is gateway for all native services like NMSLIB, Faiss , Let's give responsibility for deciding which index to intialize to respective service , in this case let's faiss service decide , I need to call binary vs normal.

I would even prefer , giving this responsibility to actually the ones who is taking action , Let's create faiss factory at cpp layer which decides what method to call. with this all the Api service layer is clean and there is no branching in service layer.

else for every function we keep checking if else and if someting else comes up , we will end up having one more else.and this changes will be at multiple places like we are having here and below functions as well.

Reply:
Resolved as this will be a change when JNI refactoring is done.